packages/chain-bridge/src/adapters/StargateAdapter.sol

```diff
@@ -288,151 +295,142 @@ contract StargateAdapter is
     }
 
     /**
-     * @dev Execute the actual sendToken call
+     * @dev Execute the actual sendToken call with consolidated logic to avoid stack too deep
      */
     function _executeSendToken(
         bytes32 operationId,
         BridgeTypes.ExecuteTransferParams memory params,
-        uint256 providedFee
+        uint256 providedFee,
+        BridgeTypes.BridgeOptions memory options
     ) internal {
+        LocalContext memory ctx;
+
         // Get the source chain Stargate contract
-        address stargateContract = assetToStargateContract[params.asset];
-        IStargateV2 stargate = IStargateV2(stargateContract);
+        ctx.stargateContract = assetToStargateContract[params.asset];
+        ctx.stargate = IStargateV2(ctx.stargateContract);
 
         // Approve Stargate contract to spend the tokens
-        IERC20(params.asset).forceApprove(stargateContract, params.amount);
+        IERC20(params.asset).forceApprove(ctx.stargateContract, params.amount);
 
         // Resolve destination adapter via registry
-        address destinationAdapter = _peerAdapter(params.destinationChainId);
-        if (destinationAdapter == address(0)) revert UnsupportedChain();
-
-        BridgeTypes.RelayedTransferParams memory atm = BridgeTypes
-            .RelayedTransferParams({
-                recipient: params.target,
-                asset: params.asset,
-                amount: params.amount,
-                sourceChainId: uint16(block.chainid),
-                operationId: operationId,
-                originator: params.originator,
-                message: params.message
-            });
+        ctx.destinationAdapter = _peerAdapter(params.destinationChainId);
+        if (ctx.destinationAdapter == address(0)) revert UnsupportedChain();
+
+        // Build relayed transfer params
+        ctx.atm = BridgeTypes.RelayedTransferParams({
+            recipient: params.target,
+            asset: params.asset,
+            amount: params.amount,
+            sourceChainId: uint16(block.chainid),
+            operationId: operationId,
+            originator: params.originator,
+            message: params.message
+        });
 
         // Build SendParam - Stargate will wrap this with OFTComposeMsgCodec internally
-        SendParam memory sendParam = _buildSendParam(
+        ctx.sendParam = _buildSendParam(
             params.destinationChainId,
-            destinationAdapter,
+            ctx.destinationAdapter,
             params.amount,
-            _encodeRelayedTransferParams(atm)
+            _encodeRelayedTransferParams(ctx.atm),
```

Comment:
it can be done inline so you skip the atm initialization, might help with stack too deep
```ts
_encodeRelayedTransferParams(
    BridgeTypes.RelayedTransferParams({
            recipient: params.target,
            asset: params.asset,
            amount: params.amount,
            sourceChainId: uint16(block.chainid),
            operationId: operationId,
            originator: params.originator,
            message: params.message
        })
)
```